docker/owlbot/java/src/convert-synth.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import ast
+import os
+import pathlib
+from typing import Union
+
+import click
+from jinja2 import Environment, FileSystemLoader
+
+
+root_directory = pathlib.Path(
```

Comment:
Please add comments and doc strings to this whole file.

Reply:
Done.